x-pack/plugins/index_management/public/hooks/use_details_page_mappings_model_management.ts

```diff
@@ -104,7 +104,7 @@ export const useDetailsPageMappingsModelManagement = () => {
         Record<string, TrainedModelStat['deployment_stats'] | undefined>
       >((acc, { model_id: modelId, deployment_stats: stats }) => {
         if (modelId && stats) {
-          acc[stats.deployment_id] = stats;
+          acc[modelId] = stats;
```

Comment:
I don't think this is correct. Semantic text models should be referenced by their deployment ID, not their model ID, because they only reference the model ID with the deployment ID that is identical to the inference endpoint name.